### ISSUE TITLE
성능 개선을 위한 학습 전략

### ISSUE THREAD

**Issue by eunseongy** [eunseongy/NLP-class-project_Platform-Prediction-for-News-Articles#2]
### 관련 커밋
- c20d0f5f500af6eeed65bec3cc0ec0bafe5e78df

데이터 특성에 맞춰 출처(referrer) 분류 체계를 단순화하고, soft label 기반 학습 방식을 적용하여 모델의 일반화 성능 및 안정성을 개선하고자 함. 또한 학습 데이터 분할 시 클래스 비율을 유지하여 학습-검증 분포 차이를 최소화함.

 
## 1. Referrer 클래스 단순화

referrer 값이 지나치게 다양한 문제가 있어 모델 학습 시 노이즈로 작용할 가능성이 큼.
따라서 다음과 같이 일부 항목을 기타(Other) 그룹으로 통합함:

```
"Facebook": "기타",
"namu.wiki": "기타",
"야후": "기타",

```

→ 불필요한 희소 클래스 제거로 class imbalance 완화 및 feature 신뢰도 향상

## 2. 주요 Referrer 추출 (Major Referrer Labeling)

각 referrer별 soft score를 계산한 뒤, 가장 높은 값을 갖는 항목을 대표 referrer 로 선택:
`
dataset_df["major_referrer"] = dataset_df[referrer_labels].idxmax(axis=1)
`
→ 이 major referrer를 기준으로 train/val을 8:2로 분할
→ 추가로, 각 referrer별 비율이 train과 val에서 동일하도록 분포를 맞춤 (Stratified split)

### 3-1. 공유 비율 정규화

`df["share_frac"] = df["share"] / 100.0
`

→ share 값을 0~1 구간으로 변환하여 모델 입력에 적합한 형태로 정규화함.

### 3-2. Soft Label 생성 (조회수 기반 가중 평균)

Soft Label 정의:

날짜별 조회수 변화와 유입 비율(share)을 함께 고려하여 referrer별 가중 확률값을 생성하는 방식

구현 코드:
```

grouped = (
    df
    .groupby(["article_id", "referrer"])
    .apply(lambda g: (g["views_total"] * g["share_frac"]).sum() / g["views_total"].sum())
    .reset_index(name="soft_label")
)
```

→ 날짜별 조회수 views_total 을 가중치로 사용
→ 유입경로 확률을 weighted soft label 로 변환
→ Hard label보다 더 부드러운 학습 신호 제공 → 일반화 성능 향상 기대
 

## 4. Loss Function — Cross Entropy with Soft LabelsSoft label

손실 함수는 다음과 같이 cross-entropy 기반으로 진행

soft label을 확률 분포로 두고 모델 출력 확률과 비교

hard one-hot보다 gradient smoothness가 증가

noisy class의 영향을 줄여 안정적인 학습 가능

### 향후 전략
이를 기반으로 하여 longformer, bigbird 등에 활용해 볼 예정이다.
